src/mosquito/job.cr

```diff
@@ -76,12 +85,29 @@ module Mosquito
 
     # Did the job run and fail?
     def failed?
-      ! succeeded?
+      !succeeded?
     end
 
     # abstract, override if desired.
     def rescheduleable?
       true
     end
+
+    # Handles throttling logic
+    private def increment : Nil
+      redis = Redis.instance
+      q = self.class.queue.config_q
+      redis.hincrby q, "executed", 1
+      config = redis.retrieve_hash q
+      return if config["limit"].to_i.zero? && config["period"].to_i.zero?
+
+      if config["executed"].to_i == config["limit"].to_i
+        next_batch = (Time.utc_now + config["period"].to_i.seconds)
+        redis.hset q, "executed", 0
+        redis.hset q, "next_batch", next_batch.epoch
+        log "#{"Execution limit reached".colorize.yellow} #{"next_batch".colorize.cyan} in #{config["period"].to_i.seconds} (at #{next_batch})"
+      end
+      redis.hset q, "last_executed", Time.utc_now.epoch
+    end
```

Comment:
The redis class here in Mosquito provides a nice wrapper to get/set an entire hash, and I think that might tighten this up a little. I'd have to tinker with it to see if my instincts are true or not, but I have a feeling that a `load_config` and `write_config` method pair would be a good place for the redis code here to live in. It'd also give a place to make default values, update config at worker reboots, etc.

Reply:
Currently it sets the defaults on the `job.cr` class, since the configs are job specific.  If we wanted to reset the config values on each reboot that would be easy.  Just depends on the intended behavior we want for it.

Reply:
I think it'd be least surprising to have it ditch previous job configs and re-initialize them on boot.   
I saw where you were writing the configs after writing this comment, and that completed the loop. Is it possible to get all the redis mangling for this in the same class? If possible that would make it easier for someone doing maintenance on this feature, so less head space required.